File: flutter17_mobile/lib/models/desktopPC.dart
import 'package:json_annotation/json_annotation.dart';
import 'product.dart';

part 'desktopPC.g.dart';

@JsonSerializable()
class DesktopPC {
  int? id;
  int? productId;

  // General Information
  String? processor;
  int? ram; // RAM size in GB
  String? storageType;
  int? storageCapacity; // Storage size in GB or TB
  String? graphicsCard;
  String? operatingSystem;

  // Physical Characteristics
  String? formFactor;
  double? weight;
  String? dimensions;

  // Connectivity
  int? usbPorts; // Number of USB ports
  bool? hasWiFi;
  bool? hasBluetooth;

  // Power Supply
  int? powerSupplyWattage; // PSU wattage

  // Cooling
  String? coolingType;

  // Additional Features
  bool? hasRGBLighting;

  DesktopPC({
    this.id,
    this.productId,
    this.processor,
    this.ram,
    this.storageType,
    this.storageCapacity,
    this.graphicsCard,
    this.operatingSystem,
    this.formFactor,
    this.weight,
    this.dimensions,
    this.usbPorts,
    this.hasWiFi,
    this.hasBluetooth,
    this.powerSupplyWattage,
    this.coolingType,
    this.hasRGBLighting,
  });

  factory DesktopPC.fromJson(Map<String, dynamic> json) =>
      _$DesktopPCFromJson(json);

  Map<String, dynamic> toJson() => _$DesktopPCToJson(this);
}

File: angular17_web/src/app/models/desktop-pc/desktopPC.ts
export class DesktopPC {
  id!: number;
  productId!: number;

  processor!: string;
  ram!: number;
  storageType!: string;
  storageCapacity!: number;
  graphicsCard!: string;
  operatingSystem!: string;

  formFactor!: string;
  weight!: number;
  dimensions!: string;

  usbPorts!: number;
  hasWiFi!: boolean;
  hasBluetooth!: boolean;

  powerSupplyWattage!: number;

  coolingType!: string;

  hasRGBLighting!: boolean;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.productId = data.productId;

    this.processor = data.processor;
    this.ram = data.ram;
    this.storageType = data.storageType;
    this.storageCapacity = data.storageCapacity;
    this.graphicsCard = data.graphicsCard;
    this.operatingSystem = data.operatingSystem;

    this.formFactor = data.formFactor;
    this.weight = data.weight;
    this.dimensions = data.dimensions;

    this.usbPorts = data.usbPorts;
    this.hasWiFi = data.hasWiFi;
    this.hasBluetooth = data.hasBluetooth;

    this.powerSupplyWattage = data.powerSupplyWattage;

    this.coolingType = data.coolingType;

    this.hasRGBLighting = data.hasRGBLighting;
  }
}
